src/openmrs-backend-dependcies.ts

```diff
@@ -0,0 +1,49 @@
+import { openmrsFetch } from "@openmrs/esm-api";
+const semver = require("semver");
+import backEndDeps from "../backEndDeps.json";
+
+export function fetchBackendDeps() {
+  return openmrsFetch(`/ws/rest/v1/module?v=custom:(uuid,version)`, {
+    method: "GET"
+  });
+}
+
+export function checkBackendDeps(dependency: any) {
+  fetchBackendDeps().then(({ data }) => {
+    let foundDependecies: any[] = [];
+    for (let [uuid] of Object.entries(dependency)) {
+      let found = false;
+      data.results.map(element => {
+        if (element.uuid === uuid) {
+          found = true;
+          foundDependecies.push(element);
+        }
+      });
+      if (!found) {
+        throw Error(`${uuid} backend end module not supported`);
```

Comment:
This error is being thrown inside of `.then()` which means that it will cause an unhandled promise rejection instead of an actual error. You should handle the promise rejection with a `.catch()` that ensures that the error gets thrown to the window appropriately